omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/AlertResource2_0.java

```diff
@@ -0,0 +1,270 @@
+	@Override
+	public List<String> getPropertiesToExposeAsSubResources() {
+		return Collections.singletonList(RECIPIENTS);
+	}
+
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl modelImpl = ((ModelImpl) super.getGETModel(rep))
+				.property("uuid", new StringProperty())
+				.property("display", new StringProperty())
+				.property(ALERT_ID, new IntegerProperty());
+
+		if (rep instanceof DefaultRepresentation) {
+			modelImpl
+					.property(TEXT, new StringProperty())
+					.property(SATISFIED_BY_ANY, new BooleanProperty())
+					.property(ALERT_READ, new BooleanProperty())
+					.property(DATE_TO_EXPIRE, new DateProperty())
+					.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientGetRef")));
+		}
+		if (rep instanceof FullRepresentation) {
+			modelImpl
+					.property(TEXT, new StringProperty())
+					.property(SATISFIED_BY_ANY, new BooleanProperty())
+					.property(ALERT_READ, new BooleanProperty())
+					.property(DATE_TO_EXPIRE, new DateProperty())
+					.property(CREATOR, new ObjectProperty())
+					.property(DATE_CREATED, new DateTimeProperty())
+					.property(CHANGED_BY, new ObjectProperty())
+					.property(DATE_CHANGED, new DateTimeProperty())
+					.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientGetRef")));
+		}
+		return modelImpl;
+	}
+
+	@Override
+	public Model getCREATEModel(Representation rep) {
+		return new ModelImpl()
+				.property(TEXT, new StringProperty())
+				.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientCreate")))
+				.property(SATISFIED_BY_ANY, new BooleanProperty())
+				.property(DATE_TO_EXPIRE, new DateProperty());
+	}
+
+	@Override
+	public Model getUPDATEModel(Representation rep) {
+		return new ModelImpl()
+				.property(TEXT, new StringProperty())
+				.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientCreate")))
+				.property(SATISFIED_BY_ANY, new BooleanProperty())
+				.property(DATE_TO_EXPIRE, new DateProperty());
+	}
+
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty(TEXT);
+		description.addRequiredProperty(RECIPIENTS);
+		description.addProperty(SATISFIED_BY_ANY);
+		description.addProperty(DATE_TO_EXPIRE);
+		return description;
+	}
+
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		return getCreatableProperties();
+	}
+
+	@Override
+	public Alert newDelegate() {
+		Alert alert = new Alert();
+		alert.setRecipients(new HashSet<>()); // to prevent NullPointerException
+		return alert;
+	}
+
+	@Override
+	public Alert getByUniqueId(String uniqueId) {
+		boolean includeRetired = true;
+		List<Alert> alerts = Context.getAlertService().getAllAlerts(includeRetired);
```

Comment:
This does not look efficient.

Reply:
There is no method to `getAlertByUuid` in `AlertService`. I could add one but then we would have to move this resource to omod-2.5, after releasing the new AlertService for OpenMRS 2.5.